ui/tests/unit/routes/vault/cluster/oidc-callback-test.js

```diff
@@ -48,6 +49,108 @@ module('Unit | Route | vault/cluster/oidc-callback', function (hooks) {
     this.callbackUrlQueryParams('');
   });
 
+  module('getParamsForCallback helper fn', function () {
+    test('it parses params correctly with regular inputs no namespace', function (assert) {
+      const params = {
+        state: 'my-state',
+        code: 'my-code',
+        path: 'oidc-path',
+      };
+      const results = getParamsForCallback(params, '?code=my-code&state=my-state');
+      assert.deepEqual(results, { source: 'oidc-callback', ...params });
+    });
+
+    test('it parses params correctly regular inputs and namespace param', function (assert) {
+      const params = {
+        state: 'my-state',
+        code: 'my-code',
+        path: 'oidc-path',
+        namespace: 'my-namespace/nested',
+      };
+      const results = getParamsForCallback(params, '?code=my-code&state=my-state');
+      assert.deepEqual(results, { source: 'oidc-callback', ...params });
+    });
+
+    test('it parses params correctly regular inputs and namespace in state', function (assert) {
+      const queryString = '?code=my-code&state=my-state,ns=blah';
+      const params = {
+        state: 'my-state,ns', // mock what Ember does with the state QP
```

Comment:
the params returned by `paramsFor` only come in like this when the queryString has an unencoded ns